src/Sprite/OP2BmpLoader.cpp

```diff
@@ -8,6 +8,9 @@
 OP2BmpLoader::OP2BmpLoader(std::string bmpFilename, std::string artFilename) :
 	bmpReader(bmpFilename), artFile(ArtFile::Read(artFilename)) { }
 
+OP2BmpLoader::OP2BmpLoader(std::string bmpFilename, Stream::BidirectionalReader& artFileStream) :
+	bmpReader(bmpFilename), artFile(ArtFile::Read(artFileStream)) { }
```

Comment:
This signature could be relaxed to take a `Stream::Reader`, of which `Stream::BidirectionalReader` is a subclass of. There appear to be no uses of the subclass behaviour. The parameter is simply passed to `ArtFile::Read`, which takes the base class type.

----

Is there a reason to take an stream for the art file, rather than an `ArtFile` instance itself? Might result in less repeated parsing if the same `ArtFile` is used multiple times. That may be relevant here as the art file is a packed format containing many independent bitmaps.

Reply:
I hadn't thought about passing in a reference to ArtFile. I was designing so a consumer could extract/load bitmaps without worrying about ArtFile.

At a minimum lets pass Stream:Reader. Good catch there.